plonk-wasm/src/rayon.rs

```diff
@@ -79,25 +81,23 @@ impl PoolBuilder {
     // Important: it must take `self` by reference, otherwise
     // `start_worker_thread` will try to receive a message on a moved value.
     pub fn build(&mut self) {
-        unsafe {
-            THREAD_POOL = Some(
-                rayon::ThreadPoolBuilder::new()
-                    .num_threads(self.num_threads)
-                    // We could use postMessage here instead of Rust channels,
-                    // but currently we can't due to a Chrome bug that will cause
-                    // the main thread to lock up before it even sends the message:
-                    // https://bugs.chromium.org/p/chromium/issues/detail?id=1075645
-                    .spawn_handler(move |thread| {
-                        // Note: `send` will return an error if there are no receivers.
-                        // We can use it because all the threads are spawned and ready to accept
-                        // messages by the time we call `build()` to instantiate spawn handler.
-                        self.sender.send(thread).unwrap_throw();
-                        Ok(())
-                    })
-                    .build()
-                    .unwrap_throw(),
-            )
-        }
+        let thread_pool_builder = rayon::ThreadPoolBuilder::new()
+            .num_threads(self.num_threads)
+            // We could use postMessage here instead of Rust channels,
+            // but currently we can't due to a Chrome bug that will cause
+            // the main thread to lock up before it even sends the message:
+            // https://bugs.chromium.org/p/chromium/issues/detail?id=1075645
+            .spawn_handler(move |thread| {
+                // Note: `send` will return an error if there are no receivers.
+                // We can use it because all the threads are spawned and ready to accept
+                // messages by the time we call `build()` to instantiate spawn handler.
+                self.sender.send(thread).unwrap_throw();
+                Ok(())
+            });
+
+        THREAD_POOL.with_borrow_mut(|pool| {
```

Comment:
Also: Consider moving the `thread_pool_builder` building process into the `with_borrow_mut(..)` maybe. Like this it can happen (can it?) that two concurrent calls to `build` will be out of order: call A, call B, building and setting pool B, building and setting pool A. Not sure if this is desirable (maybe I'm wrong assuming these two calls can be parallel at all).